overlord/state/taskrunner.go

```diff
@@ -84,6 +90,24 @@ func (r *TaskRunner) AddHandler(kind string, do, undo HandlerFunc) {
 	r.handlers[kind] = handlerPair{do, undo}
 }
 
+// AddOptionalHandler register functions for doing and undoing tasks that match
+// the given predicate if there is no handler registered for task's kind.
```

Comment:
// the given predicate if no explicit handler was registered for the task kind.

(note that there's nothing named "task" in this context, so "for task's kind" is talking about something missing)